File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        maxWater=0
        left=0
        right=len(height)-1

        while(left<right):
            width=right-left
            h=min(height[left],height[right])
            water=h*width
            maxWater=max(maxWater,water)

            if(height[left]<height[right]):
                left+=1
            else:
                right-=1
            
        return maxWater

File: 11-container-with-most-water/container-with-most-water.java
import java.util.*;
class Solution {
    public int maxArea(int[] height) {
        int maxWater=0;
        int left=0;
        int right=height.length-1;

        while(left<right){
           int width=right-left;
            int h=Math.min(height[left],height[right]);
            int water=h*width;
            maxWater=Math.max(water,maxWater);

            if(height[left]<height[right]){
                left+=1;
            }
            else{
                right-=1;
            }

        }
        return maxWater;

    }
}